### ISSUE TITLE
Unable to explore cardinality on our VMUI

### ISSUE THREAD

**Issue by surekha3** [VictoriaMetrics/VictoriaMetrics#6417]
### Is your feature request related to a problem? Please describe

We've deployed victoriametrics cluster into AWS EKS. We're experiencing high disk usage by storage pods. When dig deeper, indexdb is consuming almost 2x-3x more space than data folder. According to Victoriametrics docs, this problem might be due to high churn rate. So inorder to identify root cause, I would like to view cardinality explorer but in our VMUI, it's throwing **Error: Request failed with status 422** when I try to explore cardinality. Please help

### Describe the solution you'd like

VMUI should display cardinality

### Describe alternatives you've considered

How to fix high churn rate

**Comment by surekha3** [comment]
Hi @zekker6 ,

1. We're running victoriametrics app version 1.95.1
2. In victoriametrics select logs, I have seen attached error log. Also after sometime, I observed there is drastic consumption of CPU and RAM by victoriametrics storage pods. Is this because of reason of VMUI?
3. It's working sometimes but when I changed the date on top right it's throwing attached error page
<img width="943" alt="victoriaselectlogs" src="https://github.com/VictoriaMetrics/VictoriaMetrics/assets/49813610/d2934940-6a33-47cd-a99b-0ef274f7ff3b">
<img width="956" alt="victoriauierror" src="https://github.com/VictoriaMetrics/VictoriaMetrics/assets/49813610/d9f71f84-682c-494e-9566-6e130398fc7d">
<img width="943" alt="victoriaselectlogs" src="https://github.com/VictoriaMetrics/VictoriaMetrics/assets/49813610/aadbb897-68b9-4acf-82a2-f29c82b3c755">

**Comment by zekker6** [comment]
@surekha3 Based on the vmselect error messages it seems like vmstorage is not able to keep up with the requests rate and thus vmselect returns an error. 
I would suggest to set up a [monitoring](https://docs.victoriametrics.com/cluster-victoriametrics/#monitoring) for VictoriaMetrics cluster and a [dashboard](https://grafana.com/grafana/dashboards/11176-victoriametrics-cluster/). By using this dashboard you will be able to see if vmstorage hits the resources limits.

**Comment by surekha3** [comment]
@zekker6 You mean vmstorage pod resource limits or -search.maxconcurrentRequests parametes in victoriametrics config?

**Comment by zekker6** [comment]
@surekha3 In this case it is `search.maxconcurrentRequests`. Default value for this flag is calculated as `number of CPUs * 2` in this case. Usually this default allows to effectively utilize resources available to vmstorage node so it is likely that vmstorage is hitting resource limits as well.

**Comment by surekha3** [comment]
okay when I load VMUI explore cardinality for today's date. it's working fine. But when I load for yesterday's date it's throwing above attached 422 error. So in this case, when I increase maxconcurrentRequests parameter, will it work fine?
Also after using this VMUI, cpu and memory utilisation increased drastically. Is it because of this parameter only?

**Comment by zekker6** [comment]
> okay when I load VMUI explore cardinality for today's date. it's working fine. But when I load for yesterday's date it's throwing above attached 422 error. So in this case, when I increase maxconcurrentRequests parameter, will it work fine?

Based on the info above it should help. However, screenshot of browser requests lacks an actual request which returned 422 error. It would be great if you could open "network" tab and run a request once again just to confirm that request does return this specific error.

> Also after using this VMUI, cpu and memory utilisation increased drastically. Is it because of this parameter only?

It's hard to tell without cluster monitoring as mentioned above. Using Grafana dashboard for the cluster monitoring will allow to see which requests were taking place at this time and correlate these requests with resource usage spikes.

**Comment by zekker6** [comment]
@surekha3 You can hover the cursor over an arrow in the bottom of the graph and see the title of this vertical line. Yellow ones are usually representing a component restart. It's better to investigate the root cause of the restart in order to make sure the cluster is stable.

**Comment by surekha3** [comment]
@zekker6 you're correct. It states getting restarted. Is that because of leaving default value for -search.maxconcurrentRequests value as 2. Will heavy load causes these restarts?

**Comment by zekker6** [comment]
@surekha3 Heavy load can cause restarts only in case OOM is involved. `-search.maxConcurrentRequests` limits amount of requests vmstorage node will process at a time in order to prevent running out of resources, so this should not cause restarts.
Could you check the reason of pods restart in k8s events?

**Comment by surekha3** [comment]
@zekker6 Though it's showing as pod restarts in Grafana dashboard, when I ran kubectl events command, it's not showing as pod get restarted which seems strange

**Comment by zekker6** [comment]
@surekha3 Restarts annotation uses the following query: `sum(changes(vm_app_start_timestamp)) by(job)`
Could you run this query and check the value of `vm_app_start_timestamp`?

**Comment by makasim** [comment]
Closing this issue due to age and lack of additional information. Unfortunately, we’re unable to proceed without the necessary details. Please feel free to reopen or comment if you have any updates or further questions.
